### PR TITLE
Add Diff commands for Numstat, Raw, and Patch diffs

### LINKED ISSUE #973: Add support for combined/merge diffs (--cc format)
## Summary

The new diff commands (`Git::Commands::Diff::Numstat`, `Raw`, `Patch`) and `DiffParser` module currently only support standard two-way diffs. Combined/merge diffs (e.g., `git diff --cc`, `git show <merge>`) use a different format and are not parsed.

## Background

Combined diffs are used to show how merge conflicts were resolved. They have a fundamentally different structure:
- Multiple columns of `+`/`-` markers (one per parent commit)
- Multiple `@@` markers in hunk headers
- Used primarily with `git show <merge-commit>` or `git log -p` on merge commits

Example combined diff format:
```diff
diff --cc file.txt
index abc123,def456..789ghi
--- a/file.txt
+++ b/file.txt
@@@ -1,5 -1,5 +1,5 @@@
  common line
- removed from first parent
 -removed from second parent
++resolved line
  common line
```

## Implementation Notes

Adding support would require:
1. A new `--cc` option in the Patch command (or a new CombinedPatch command)
2. A new `CombinedPatch` submodule in `DiffParser` to handle the multi-parent format
3. New info classes to represent multiple-parent diff hunks (e.g., `DiffFileCombinedInfo`)
4. Unit and integration tests with merge commit scenarios

## Current State

The limitation is documented in:
- `lib/git/commands/diff.rb` (module documentation)
- `lib/git/diff_parser.rb` (parser documentation)

## Priority

Low - standard two-way diffs cover the primary use cases. This is primarily needed for tools that analyze merge conflict resolutions.

### PR DESCRIPTION
**Pull request by jcouball**
## Summary

Implement `Git::Commands::Diff::{Numstat,Raw,Patch}` commands using the new command architecture with `raise_on_failure: false` pattern to handle git diff's exit codes correctly:
- Exit code 0: no differences found
- Exit code 1: differences found (success)
- Exit code ≥2: actual error

## Features

- Parse numstat, raw, and patch diff formats into immutable data objects
- Support all diff options: `cached`/`staged`, `merge_base`, `no_index`, `pathspecs`
- Handle renames, copies, binary files, type changes, and submodules
- Support `dirstat` output for directory-level change statistics
- Properly unescape git's quoted paths (spaces, tabs, UTF-8 encoded as octal)

## Data Objects Added

| Class | Description |
|-------|-------------|
| `DiffResult` | Aggregate result with files array and summary statistics |
| `DiffFileNumstatInfo` | Line insertion/deletion counts per file |
| `DiffFileRawInfo` | File mode, SHA, status, and similarity information |
| `DiffFilePatchInfo` | Full unified diff patch text per file |
| `FileRef` | Immutable file reference (mode, SHA, path) |
| `DirstatInfo` | Directory-level change percentages |

## Test Coverage

- **189 tests total** (113 unit + 76 integration)
- Edge cases covered:
  - Rename/copy detection with similarity percentages
  - Binary files (including GIT binary patch format)
  - Type changes (file ↔ symlink)
  - Submodule add/update
  - Special character paths (spaces, tabs, UTF-8, quotes)
  - Mode changes (executable bit)

## Usage Examples

```ruby
# Get patch diff between two commits
result = Git::Commands::Diff::Patch.new(repo.lib).call('main', 'feature')
result.files.each do |file|
  puts "#{file.path}: +#{file.insertions}/-#{file.deletions}"
  puts file.patch if file.patch
end

# Get numstat (line counts only, faster)
result = Git::Commands::Diff::Numstat.new(repo.lib).call('HEAD~5', 'HEAD')
puts "Total: +#{result.total_insertions}/-#{result.total_deletions}"

# Get raw diff with file metadata
result = Git::Commands::Diff::Raw.new(repo.lib).call(cached: true)
result.files.select(&:renamed?).each do |file|
  puts "#{file.src_path} → #{file.path} (#{file.similarity}% similar)"
end
```

## Related

- Closes #973 (documents combined/merge diffs as out of scope for future consideration)

### CODE REVIEW

File: lib/git/commands/diff/raw.rb [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+        #   @param options [Hash] command options
+        #
+        #   @option options [Array<String>] :pathspecs (nil) zero or more pathspecs to limit diff to
+        #
+        # @return [Git::DiffResult] diff result with per-file raw info
+        #
+        # @raise [Git::FailedError] if git returns exit code >= 2 (actual error)
+        #
+        def call(commit1 = nil, commit2 = nil, dirstat: nil, pathspecs: nil, **)
+          args = ARGS.build(commit1, commit2, dirstat: dirstat, **)
```
[the remark this answers is not in the document]
Comment (jcouball) [reply lib/git/commands/diff/raw.rb]: Changed Raw#call and Patch#call signature as requested.

File: lib/git/commands/diff/patch.rb [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+        #   @param options [Hash] command options
+        #
+        #   @option options [Array<String>] :pathspecs (nil) zero or more pathspecs to limit diff to
+        #
+        # @return [Git::DiffResult] diff result with per-file patch information
+        #
+        # @raise [Git::FailedError] if git returns exit code >= 2 (actual error)
+        #
+        def call(commit1 = nil, commit2 = nil, dirstat: nil, pathspecs: nil, **)
+          args = ARGS.build(commit1, commit2, dirstat: dirstat, **)
```
[the remark this answers is not in the document]
Comment (jcouball) [reply lib/git/commands/diff/patch.rb]: Changed method signature as suggested.
